orb-ui/cone/examples/cone-simulation.rs

```diff
@@ -0,0 +1,188 @@
+use orb_rgb::Argb;
+
+const CONE_LED_STRIP_DIMMING_DEFAULT: u8 = 10_u8;
+const CONE_SIMULATION_UPDATE_PERIOD_S: u64 = 2;
+const CONE_LED_STRIP_MAXIMUM_BRIGHTNESS: u8 = 20;
+
+#[tokio::main]
+async fn main() -> eyre::Result<()> {
+    let registry = tracing_subscriber::registry();
+    #[cfg(tokio_unstable)]
+    let registry = registry.with(console_subscriber::spawn());
+    registry
+        .with(fmt::layer())
+        .with(
+            EnvFilter::builder()
+                .with_default_directive(LevelFilter::INFO.into())
+                .from_env_lossy(),
+        )
+        .init();
+
+    let devices = ftdi_embedded_hal::libftd2xx::list_devices()?;
+    for device in devices.iter() {
+        tracing::debug!("Device: {:?}", device);
+    }
+
+    loop {
+        let (tx, mut rx) = broadcast::channel(10);
+        if let Ok((mut cone, cone_handles)) = orb_cone::Cone::spawn(tx) {
+            // spawn a thread to receive events
+            let button_listener_task: JoinHandle<eyre::Result<()>> =
+                task::spawn(async move {
+                    let mut button_state = ButtonState::Released;
+                    loop {
+                        match rx.recv().await {
+                            Ok(event) => match event {
+                                ConeEvent::Button(state) => {
+                                    if state != button_state {
+                                        tracing::info!("🔘 Button {:?}", state);
+                                        button_state = state;
+                                    }
+                                }
+                                ConeEvent::Cone(state) => {
+                                    tracing::info!("🔌 Cone {:?}", state);
+                                }
+                            },
+                            Err(RecvError::Closed) => {
+                                return Err(eyre!(
+                                    "Cone events channel closed, cone disconnected?"
+                                ))
+                            }
+                            Err(RecvError::Lagged(skipped)) => {
+                                tracing::warn!("🚨 Skipped {} cone events", skipped);
+                            }
+                        }
+                    }
+                });
+
+            // create one shot to gracefully terminate simulation
+            let (kill_sim_tx, mut kill_sim_rx) = tokio::sync::oneshot::channel::<()>();
+            let simulation_task: JoinHandle<eyre::Result<()>> = tokio::task::spawn(
+                async move {
+                    let mut counter = 0;
+                    loop {
+                        tokio::select! {
+                            _ = &mut kill_sim_rx => {
+                                return Ok(());
+                            }
+                            _ = tokio::time::sleep(std::time::Duration::from_secs(CONE_SIMULATION_UPDATE_PERIOD_S)) => {
```

Comment:
nit: reduce indents by making the stuff after => happen *after* the select is finished. Same concept as doing 
```rust
if is_bad {
  bail!("oh no");
}
do_good();
```

rather than
```rust
if is_bad {
  bail!("oh no");
} else {
  do_good()
}
```